File: api/src/main/java/org/delivery/api/domain/store/converter/StoreConverter.java
package org.delivery.api.domain.store.converter;

import lombok.RequiredArgsConstructor;
import org.delivery.api.common.annotation.Converter;
import org.delivery.api.common.error.ErrorCode;
import org.delivery.api.common.exception.ApiException;
import org.delivery.api.domain.store.controller.model.StoreRegisterRequest;
import org.delivery.api.domain.store.controller.model.StoreResponse;
import org.delivery.db.store.StoreEntity;

import java.util.Optional;

@Converter
@RequiredArgsConstructor
public class StoreConverter {

    public StoreEntity toEntity(
            StoreRegisterRequest request
    ){

        return Optional.ofNullable(request)
                .map(it -> {
                    return StoreEntity.builder()
                            .name(it.getName())
                            .address(it.getAddress())
                            .category(it.getCategory())
                            .thumbnailUrl(it.getThumbnailUrl())
                            .minimumAmount(it.getMinimumAmount())
                            .minimumDeliveryAmount(it.getMinimumDeliveryAmount())
                            .phoneNumber(it.getPhoneNumber())
                            .build();
                })
                .orElseThrow(() -> new ApiException(ErrorCode.NULL_POINT));
    }
    public StoreResponse toResponse(
            StoreEntity entity
    ){

        return Optional.ofNullable(entity)
                .map(it -> {
                    return StoreResponse.builder()
                            .id(it.getId())
                            .name(it.getName())
                            .address(it.getAddress())
                            .category(it.getCategory())
                            .status(it.getStatus())
                            .star(it.getStar())
                            .thumbnailUrl(it.getThumbnailUrl())
                            .minimumAmount(it.getMinimumAmount())
                            .minimumDeliveryAmount(it.getMinimumDeliveryAmount())
                            .phoneNumber(it.getPhoneNumber())
                            .build();
                })
                .orElseThrow(() -> new ApiException(ErrorCode.NULL_POINT));
    }
}

File: api/src/main/kotlin/org/delivery/api/domain/store/converter/StoreConverter.kt
package org.delivery.api.domain.store.converter

import org.delivery.api.domain.store.controller.model.StoreRegisterRequest
import org.delivery.api.domain.store.controller.model.StoreResponse
import org.delivery.common.annotation.Converter
import org.delivery.db.store.StoreEntity

@Converter
class StoreConverter {

    fun toEntity(
        request : StoreRegisterRequest?
    ):StoreEntity?{
        return StoreEntity(
            name = request?.name,
            address = request?.address,
            category = request?.category,
            thumbnailUrl = request?.thumbnailUrl,
            minimumAmount = request?.minimumAmount,
            minimumDeliveryAmount = request?.minimumDeliveryAmount,
            phoneNumber = request?.phoneNumber
        )
    }


    fun toResponse(
            entity: StoreEntity?
    ): StoreResponse {
        return StoreResponse(
            id = entity?.id,
            name = entity?.name,
            address = entity?.address,
            status = entity?.status,
            category = entity?.category,
            star = entity?.star,
            thumbnailUrl = entity?.thumbnailUrl,
            minimumAmount = entity?.minimumAmount,
            minimumDeliveryAmount = entity?.minimumDeliveryAmount,
            phoneNumber = entity?.phoneNumber
        )
    }
}